mod/invitefriends/views/default/forms/invitefriends/invite.php

```diff
@@ -2,47 +2,35 @@
 
 /**
  * Elgg invite form contents
- *
- * @package ElggInviteFriends
  */
 
-$site = elgg_get_site_entity();
-$default_message = elgg_echo('invitefriends:message:default', array($site->name));
+$default_message = elgg_echo('invitefriends:message:default', [elgg_get_site_entity()->getDisplayName()]);
 
-$introduction = elgg_echo('invitefriends:introduction');
+echo elgg_view('output/longtext', [
+	'value' => elgg_echo('invitefriends:introduction'),
+]);
 
-$emails_label = elgg_echo('invitefriends:emails');
-$message_label = elgg_echo('invitefriends:message');
-
-$emails = elgg_get_sticky_value('invitefriends', 'emails');
-$message = elgg_get_sticky_value('invitefriends', 'emailmessage', $default_message);
-
-$emails_textarea = elgg_view('input/plaintext', array(
+echo elgg_view_field([
+	'#type' => 'plaintext',
+	'#label' => elgg_echo('invitefriends:emails'),
 	'id' => 'invitefriends-emails',
 	'name' => 'emails',
-	'value' => $emails,
+	'value' => elgg_get_sticky_value('invitefriends', 'emails'),
```

Comment:
Well, we did the same with lists, in the end we ended up moving list rendering into `listing/` views. I'd rather we keep resource elements separate, so it's easy to replace the form/add vars without hacking the resource view.